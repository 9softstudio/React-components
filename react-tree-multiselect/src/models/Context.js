import React from 'react';

export default React.createContext({
    rootNode: {},
    nodes: [],
    onToggle: () => { },
    onCheck: () => { }
});