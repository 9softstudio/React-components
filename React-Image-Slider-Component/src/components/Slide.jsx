import React, { Component } from 'react';
import Context from '../slider-context';
import { mergeClassName } from '../utils';

export default class Slide extends Component {
    render() {
        const { imageInfo, index } = this.props;
        const extensionOfImage = imageInfo.url.substring(imageInfo.url.lastIndexOf('.'));
        return (
            <Context.Consumer>{
                ({ onImageClick }) =>
                <div className={mergeClassName(this.props, 'slide')} onClick={() => onImageClick(index)}>
                    <img alt={imageInfo.text} src={imageInfo.url} onError={e => e.target.src = imageInfo.url.replace(extensionOfImage, '.jpg')} />
                    <h1>{imageInfo.text}</h1>
                </div>
            }</Context.Consumer>);
    }
};