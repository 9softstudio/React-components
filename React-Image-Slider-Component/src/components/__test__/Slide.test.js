import React from 'react';
import Slide from '../Slide';
import Context from '../../slider-context';

import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';

configure({ adapter: new Adapter() });

describe('<Slide />', () => {
    test('renders correctly', () => {
        const component = renderer
            .create(
                <Context.Provider value={{onImageClick: jest.fn()}}>
                    <Slide index={2} imageInfo={{ url: "fake-url", text: "fake-text" }} style="fake-object-style" />
                </Context.Provider>
            );
        expect(component.toJSON()).toMatchSnapshot();
    });

    test.skip('click on indicator: onImageClick() should be called with index correctly', () => {
        // arrange
        const index = 999;
        const onImageClick = jest.fn();
        
        const context = <Context.Provider value={({ onImageClick: onImageClick })}></Context.Provider>;
        const wrapper = shallow(<Slide index={index} imageInfo={{ url: "fake-url", text: "fake-text" }}/>, {context});
       
        console.log(wrapper.instance());
        console.log(wrapper.find('div'));
        // action
        wrapper.simulate('click');

        // assert
        expect(onImageClick).toBeCalledWith(index);        
    });
});