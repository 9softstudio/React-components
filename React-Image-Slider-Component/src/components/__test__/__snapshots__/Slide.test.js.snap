// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Slide /> renders correctly 1`] = `
<div
  className="slide"
  onClick={[Function]}
>
  <img
    alt="fake-text"
    onError={[Function]}
    src="fake-url"
  />
  <h1>
    fake-text
  </h1>
</div>
`;
